test/e2e/specs/editor/various/convert-block-type.spec.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Code block', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should convert to a preformatted block', async ( {
+		page,
+		editor,
+	} ) => {
+		const code = 'print "Hello Dolly!"';
+
+		await editor.insertBlock( { name: 'core/code' } );
+		await page.keyboard.type( code );
+
+		// Verify the content starts out as a Code block.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
```

Comment:
Nit: Probably will be better to get rid of the snapshot and just inline the expected result here:

```suggestion
		await expect.poll( editor.getEditedPostContent )
			.toBe( `<!-- wp:code -->
<pre class="wp-block-code"><code>${code}</code></pre>
<!-- /wp:code -->` );
```